src/http/client.rs

```diff
@@ -269,13 +270,29 @@ impl Http {
 
     #[cfg(feature = "unstable_discord_api")]
     pub fn new_with_token_application_id(token: &str, application_id: u64) -> Self {
-        let mut base = Self::new_with_token(token);
+        let base = Self::new_with_token(token);
 
-        base.application_id = application_id;
+        base.set_application_id(application_id);
 
         base
     }
 
+    #[cfg(feature = "unstable_discord_api")]
+    pub fn application_id(&self) -> Option<u64> {
```

Comment:
All 25 callers of this function have `.ok_or(HttpError::ApplicationIdMissing)`. Maybe this function should return `Result<u64, HttpError>` directly?

Reply:
I'm not sure if that makes sense from a user perspective since the method doesn't return any other error variants, maybe a new private method would fit better?

Reply:
Good idea. Name suggestions: `try_application_id`, `application_id_or_err`